### PR TITLE
Login Page, Signup Page and Forgot Password Page

### PR DESCRIPTION
**Pull request by larrywang0701**
# Add Login Page, Signup Page and Forgot Password Page
## Login Page (with example input)
image
## Signup Page (with example input)
(currently only implemented partial communication logic to backend since signup backend is not implemented yet)
image
## Forgot Password Page (with example input)
image
#2 #6 #12

### CODE REVIEW

File: frontend/src/components/common/PageHeader.tsx
Comment (brein62) [inline frontend/src/components/common/PageHeader.tsx:10]:
```diff
- export default function PageHeader({isLoggedIn = true}) {
+ export default function PageHeader({ isLoggedIn = true } : { isLoggedIn : boolean }) {
```

**Review by brein62 — COMMENTED** [review]
just a minor suggestion

File: frontend/src/components/common/PageHeader.tsx
Comment (larrywang0701) [reply frontend/src/components/common/PageHeader.tsx:10]: got it

**Comment by brein62** [comment]
just wondering: is any of the login/signup/forgot password implemented yet as of now?

File: frontend/src/components/user-service/signup-page/SignupForm.tsx
Comment (brein62) [inline frontend/src/components/user-service/signup-page/SignupForm.tsx:108]: interesting algorithm for passwordStrength but.... `1234567812345678123456781234567812345678` is considered "secure" as well

File: frontend/src/components/user-service/signup-page/SignupButton.tsx
Comment (brein62) [inline frontend/src/components/user-service/signup-page/SignupButton.tsx:10]: - would be nice to use shadcn `<Button />` component instead of native `<button />`
- the arrow icon is right beside the signup text which doesn't look too great. would suggest adding more padding around the button and adding a gap between the button and text. something like
```tsx
<Button onClick={onClick} className="border bg-orange-300 hover:bg-orange-300/75 rounded-lg px-4 py-1.5 flex items-center gap-1">
  <ArrowRightIcon />
  <div>Sign up</div>
</Button>
```

File: frontend/src/components/user-service/login-page/LoginButton.tsx
Comment (brein62) [inline frontend/src/components/user-service/login-page/LoginButton.tsx:10]: refer to my suggestion for `SignupButton.tsx`

File: frontend/src/components/user-service/forgot-password-page/ForgotPasswordButton.tsx
Comment (brein62) [inline frontend/src/components/user-service/forgot-password-page/ForgotPasswordButton.tsx:10]: refer to my suggestion for `SignupButton.tsx`

**Review by brein62 — COMMENTED** [review]
won't comment too much on the user service integration logic since you're working on it, but here is a suggestion for the design of the buttons

**Comment by larrywang0701** [comment]
If I didn't understand your word wrongly, I think currently in this main repo, the login page is just a dummy page, and the signup page and forgot password page is empty (has no functionality)
https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g36/blob/main/frontend/src/pages/user-service/LoginPage.tsx
https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g36/blob/main/frontend/src/pages/user-service/SignupPage.tsx
https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g36/blob/main/frontend/src/pages/user-service/ForgotPasswordPage.tsx

And for this pull request, `LoginPage` and `ForgotPasswordPage` is fully implemented and `SignupPage` is implemented without communication to the backend

File: frontend/src/components/user-service/signup-page/SignupForm.tsx
Comment (larrywang0701) [reply frontend/src/components/user-service/signup-page/SignupForm.tsx:108]: For this I think it would be a tricky problem, as different people may consider "strong password" with different ways/aspect. For your `1234567812345678123456781234567812345678` I think it would be secure since it's really long (although its consecutive). Also your `1234567812345678123456781234567812345678` is not a valid password since it's missing upper case letter and lower case letter. As the users themselves should be more responsible on their own password's strength, for now I think my algorithm is sufficient for simple brute force hack (along with failed login attempt check and account locking). If you think this is still not enough, you can feel free to modify my algorithm after merging this pull request. Thanks

File: frontend/src/components/user-service/login-page/LoginButton.tsx
Comment (larrywang0701) [reply frontend/src/components/user-service/login-page/LoginButton.tsx]: got it

File: frontend/src/components/user-service/forgot-password-page/ForgotPasswordButton.tsx
Comment (larrywang0701) [reply frontend/src/components/user-service/forgot-password-page/ForgotPasswordButton.tsx]: got it

File: frontend/src/components/user-service/signup-page/SignupButton.tsx
Comment (larrywang0701) [reply frontend/src/components/user-service/signup-page/SignupButton.tsx]: got it

File: frontend/src/components/user-service/signup-page/SignupForm.tsx
Comment (brein62) [reply frontend/src/components/user-service/signup-page/SignupForm.tsx:118]: i think just add on an additional check for unsafe password like this, so like instead of showing the password security thing perhaps have some kind of message saying that the criteria for strong password is not met

Comment (brein62) [reply frontend/src/components/user-service/signup-page/SignupForm.tsx:118]: while for valid passwords continue what you've been doing

Comment (larrywang0701) [reply frontend/src/components/user-service/signup-page/SignupForm.tsx:118]: It already has a notification of "invalid password" if certain criteria of the password is not met, so I think adding a notification again is redundant. Since users should have their own judges on whether a password is strong or not (I mean, password like `12345678...` is really easy to be judged by the user as an unsafe password despite what my algorithm shows). So I think a user-friendly general and responsive password strength checker would be better (what I mean by "general" does not include the special cases such as your `12345678...`. Again, users should be responsible for their own password's strength at the first time, our UI just provide a reference to them but not a definite result)

**Comment by larrywang0701** [comment]
since milestone deadline is approaching, anymore reviews regarding this PR? if not can merge and close related issues, thanks

**Comment by brein62** [comment]
scope for milestone 3 is just containerisation of user service (the backend) and question service (backend) if you've seen in the chat so the current improvements here are just for future milestones, feel like they dw to bully us too much during midterm season

since we almost got penalised during the milestone 2 submission for doing user service early

however, this is very much in scope for milestone 4 so do keep working on it! will adjust the milestone deadlines accordingly

**Comment by larrywang0701** [comment]
So you mean will merge this pr after captcha logic and signup API logic on the backend side is totally done and then I finish frontend logic according to the backend logic? If so seems that I'll wait to continue my work and merge this PR until backend logic is done. Thanks

**Comment by brein62** [comment]
> So you mean will merge this pr after captcha logic and signup API logic on the backend side is totally done and then I finish frontend captcha logic and signup API logic according to the backend logic? If so seems that I'll wait to continue my work and merge this PR until the corresponding backend logic is done. Thanks

it's based on the given milestone requirements but since all of us are busy with midterms we also can't exactly work on the features now so yea do wait for now

**Comment by larrywang0701** [comment]
I think we can merge this for now as can finally have a formal user service UI rather than the dummy one. For the parts that haven't implemented maybe can implement later (e.g. captcha)

**Review by wjayee — APPROVED** [review]
LGTM! honestly it is a bit long so I just glanced through haha but we can just fix any bugs in future prs

File: frontend/src/components/user-service/signup-page/SignupForm.tsx
Comment (brein62) [reply frontend/src/components/user-service/signup-page/SignupForm.tsx:118]: i think that i'll leave it as a point of discussion to the group later since i think your implementation should be ok as is, just that i got a few suggestions to change them to account for such extreme cases. stay tuned on tele later 😃

**Review by brein62 — APPROVED** [review]
LGTM! @larrywang0701 if the current implementation works **with the backend** you may merge alr (unable to test rn). the design and changes are pretty good so i won't complain about it for now. any improvements we identify we'll deal with as separate issues.

**Pull request merged**
